packages/dev/core/src/Gizmos/scaleGizmo.ts

```diff
@@ -117,13 +125,15 @@ export class ScaleGizmo extends Gizmo implements IScaleGizmo {
     public set attachedMesh(mesh: Nullable<AbstractMesh>) {
         this._meshAttached = mesh;
         this._nodeAttached = mesh;
-        [this.xGizmo, this.yGizmo, this.zGizmo, this.uniformScaleGizmo].forEach((gizmo) => {
+        const allGizmos = this.allGizmos;
+        for (let i = 0; i < 4; i++) {

```

Comment:
Please use length instead as it is a maintenance nightmare f we add a new one in the list. Agree accessing length is slower but it is definitely a premature optimization.

If you really want you can store the length when you create the array.

This is valid for all the loop changed with hardcoded values.

Reply:
Changed, please check